applications/nrf5340_audio/src/audio_module/src/aobj_api.h

```diff
@@ -0,0 +1,494 @@
+	 *
+	 * @return 0 if successful, error otherwise
+	 */
+	int (*close)(struct handle *handle);
+
+	/**
+	 * @brief Function to reconfigure a module.
+	 *
+	 * @param handle		 A handle to the module instance
+	 * @param configuration  Pointer to the module's configuration to set
+	 *
+	 * @return 0 if successful, error otherwise
+	 */
+	int (*configuration_set)(struct handle *handle, struct amod_configuration *configuration);
+
+	/**
+	 * @brief Function to get the configuration of a module.
+	 *
+	 * @param handle		 A handle to the module instance
+	 * @param configuration  A pointer to the module's current configuration
+	 *
+	 * @return 0 if successful, error value
+	 */
+	int (*configuration_get)(struct handle *handle, struct amod_configuration *configuration);
+
+	/**
+	 * @brief Start a module.
+	 *
+	 * @param handle  The handle for the module to start
+	 *
+	 * @return 0 if successful, error otherwise
+	 */
+	int (*start)(struct handle *handle);
+
+	/**
+	 * @brief Stop a module.
+	 *
+	 * @param handle  The handle for the module to stop
+	 *
+	 * @return 0 if successful, error otherwise
+	 */
+	int (*stop)(struct handle *handle);
+
+	/**
+	 * @brief The core data processing function for the module. Can be either an
+	 *		  input, output or processor module type.
+	 *
+	 * @param handle	A handle to the module instance
+	 * @param block_tx  Pointer to the input audio block or NULL for an input module
+	 * @param block_rx  Pointer to the output audio block or NULL for an output module
+	 *
+	 * @return 0 if successful, error otherwise
+	 */
+	int (*data_process)(struct handle *handle, struct ablk_block *block_tx,
+			    struct ablk_block *block_rx);
+};
+
+/**
+ * @brief A module's minimum description.
+ */
+struct amod_description {
+	/* The module base name */
+	char *name;
+
+	/* The module type */
+	enum amod_type type;
+
+	/* A pointer to the functions in the module */
+	const struct amod_functions *functions;
+};
+
+/**
+ * @brief Module's thread configuration structure.
+ */
+struct amod_thread_configuration {
+	/* Thread stack */
+	k_thread_stack_t *stack;
+
+	/* Thread stack size */
+	size_t stack_size;
+
+	/* Thread priority */
+	int priority;
+
+	/* Pointer to a module's data receiver FIFO */
+	struct data_fifo *msg_rx;
+
+	/* Pointer to a module's data transmitter FIFO */
+	struct data_fifo *msg_tx;
+
+	/* Pointer to the data buffer slab */
+	struct k_mem_slab *data_slab;
+
+	/* Size of each memory block in bytes that will be
+	 * taken from the data buffer slab
+	 */
+	size_t data_size;
+};
+
+/**
+ * @brief Module's generic set-up structure.
+ */
+struct amod_parameters {
+	/* The module's private description */
+	struct amod_description *description;
+
+	/* The module's thread setting */
+	struct amod_thread_configuration thread;
+};
+
+/**
+ * @brief Private module handle.
+ */
+struct amod_handle {
+	/* Unique name of this module instance */
+	char name[CONFIG_AMOD_NAME_SIZE];
```

Comment:
Is there a mix here now in names? THe file is called aobj. but there are a lot of amod's in here.

Reply:
I have no idea what happened there!
The amod_api.h is now back as it should be and the aobj_api.h has been removed (renamed to be precise as ablk_api.h)